python/cudf/cudf/_lib/column.pyx

```diff
@@ -428,6 +438,47 @@ cdef class Column:
             offset,
             children)
 
+    cpdef pylibcudf.Column to_pylibcudf(self):
+        # TODO: Categoricals will need to be treated differently eventually.
+        # There is no 1-1 correspondence between cudf and libcudf for
+        # categoricals due to the ordering question.
+        if is_categorical_dtype(self.dtype):
+            col = self.base_children[0]
+        else:
+            col = self
+
+        cdef pylibcudf.DataType dtype = dtype_to_pylibcudf_type(col.dtype)
+
+        cdef pylibcudf.gpumemoryview data = None
+        if col.base_data is not None:
+            data = pylibcudf.gpumemoryview(col.base_data)
+
+        cdef pylibcudf.gpumemoryview mask = None
+        if self.nullable:
+            # TODO: Are we intentionally use self's mask instead of col's?
+            # Where is the mask stored for categoricals?
+            mask = pylibcudf.gpumemoryview(self.base_mask)
+
+        cdef Column child_column
+        children = []
+        if col.base_children:
+            for child_column in col.base_children:
+                children.append(child_column.to_pylibcudf())
+
+        return pylibcudf.Column(
+            dtype,
+            self.size,
+            data,
+            mask,
+            # TODO: We may need to do some extra work to ensure that
+            # self.null_count is not None. However, that would be out of scope
+            # for this PR and would probably need to be resolved as part of the
+            # UNKNOWN_NULL_COUNT removal.
```

Comment:
Honestly I think we can just delete this comment without adding an assert. Python doesn't optimize away asserts unless it's run with `-O`, and we can make this assumption fairly safely at this point (otherwise we would be seeing test failures all over the place after the `UNKNOWN_NULL_COUNT` removal. Unless that makes you particularly uneasy I'd rather not add the assert (but if you do want to add, there's probably a better place to put it than `to_pylibcudf` since we'd want to assert this property more generally anywhere the null count is queried I assume).